packages/js/product-editor/src/components/variations-table/variations-table.tsx

```diff
@@ -116,6 +143,8 @@ export function VariationsTable() {
 			</div>
 		);
 	}
+	// this prevents a weird jump from happening while changing pages.
+	const variations = latestVariations || lastVariations.current;
```

Comment:
As mentioned in the comment, this prevents a weird jump from happening when switching pages.
Now we just show the loading spinner overlay until the new page is loaded, instead of removing everything and showing the loading spinner.